creator-node/src/fileManager.js

```diff
@@ -748,6 +750,89 @@ async function removeFile(storagePath) {
   }
 }
 
+/**
+ * Adds path to redis set for every file of the given user and then deletes each path from disk.
+ * Uses pagination to avoid loading all files in memory for users with a lot of data.
+ * @param {string} walletPublicKey the wallet of the user to delete all data for
+ * @param {bunyan.Logger} logger the logger to print messages to
+ * @return number of files deleted
+ */
+async function deleteAllCNodeUserDataFromDisk(walletPublicKey, logger) {
+  const FILES_PER_QUERY = 10_000
+  const redisSetKey = `filePathsToDeleteFor${walletPublicKey}`
+  await redisClient.del(redisSetKey)
+
+  const cnodeUser = await models.CNodeUser.findOne({
+    where: { walletPublicKey }
+  })
+  if (!cnodeUser) throw new Error('No cnodeUser found')
+
+  const { cnodeUserUUID } = cnodeUser
+  logger.info(
+    `Fetching data to delete from disk for cnodeUserUUID: ${cnodeUserUUID}`
+  )
+
+  try {
+    // Add file paths to redis
+    const numFilesToDelete = await models.File.count({
+      where: { cnodeUserUUID }
+    })
+    let files = []
+    // Paginate on multihash because we have an index on it. Start at the lowest real character (space)
+    let prevMultihash = ' '
+    for (let i = 0; i < numFilesToDelete; i += FILES_PER_QUERY) {
+      files = await models.File.findAll({
+        attributes: ['multihash', 'dirMultihash', 'storagePath'],
+        order: [['multihash', 'ASC']],
+        where: {
+          cnodeUserUUID,
+          multihash: {
+            [models.Sequelize.Op.gt]: prevMultihash
+          }
+        },
+        limit: FILES_PER_QUERY
+      })
+
+      if (files?.length) {
+        // Save the file paths to a redis set
+        // TODO: Might need to prepend file.dirMultihash for nested paths?
```

Comment:
yeah good callout to test dirCID flow

Reply:
I looked into it and I think we don't need to check dirMultihash because our streaming logic always streams directly from the db's `storagePath`. even if it checks dirCID for verification the path that's read from is always `path.normalize(storagePath)`. updating to reflect that

Reply:
ok dope - sounds good (pls test this flow - locally when images are uploaded for a user, just confirm that they are also deleted)

Reply:
updated PR description with testing steps for the route I added ✅